clang/lib/Sema/SemaChecking.cpp

```diff
@@ -4570,7 +4570,8 @@ ExprResult Sema::BuildAtomicExpr(SourceRange CallRange, SourceRange ExprRange,
   }
 
   // Pointer to object of size zero is not allowed.
-  if (Context.getTypeInfoInChars(AtomTy).Width.isZero()) {
+  if (!AtomTy->isIncompleteType() &&
```

Comment:
If you call `RequireCompleteType()` before this `if` instead of checking for an incomplete type, do you get better diagnostic behavior?

Reply:
The current diagnostic message only says we need trivially-copyable type here, and the `RequireCompleteType` version would say we need complete type here and it sounds more understandable to me.